code/modules/research/designs.dm

```diff
@@ -43,11 +43,49 @@ other types of metals and chemistry for reagents).
 	var/lathe_time_factor = 1			//How many times faster than normal is this to build on the protolathe
 	var/dangerous_construction = FALSE	//notify and log for admin investigations if this is printed.
 	var/departmental_flags = NONE			//bitflags for deplathes.
+	var/list/datum/techweb_node/unlocked_by = list()

```

Comment:
Will however likely cause gc failures if node destroy proc doesn't properly unregister them from designs. I normally use IDs and not direct references for a reason :p (that being said IDs are harder to manage and require checking for nulls more, I'm not sure if a failed gc/hard deletion will make an entry here null or remove it.)